### PR TITLE
Update after own review

### PR DESCRIPTION
**Pull request by pbougue**
Tried to separate commits between easy (just renaming and so) and more controversial decision (so it's easier to revert).
More complicated changes should be in another PR.

### CODE REVIEW

File: source/jormungandr/tests/journey_common_tests.py [hunk 1]
```diff
@@ -1787,24 +1787,25 @@ def test_timeframe_duration_with_min_nb_journeys(self):
                 'min_nb_journeys={min_nb_journeys}&'
                 'timeframe_duration={timeframe_duration}&').format( _from='stop_area:sa1',
                                                                     to='stop_area:sa3',
-                                                                    datetime="20180315T080000",
-                                                                    min_nb_journeys="20",
+                                                                    datetime='20180315T080000',
+                                                                    min_nb_journeys='20',
```
Comment (TeXitoi) [inline source/jormungandr/tests/journey_common_tests.py]: no quote

File: source/jormungandr/tests/journey_common_tests.py [hunk 2]
```diff
@@ -1787,24 +1787,25 @@ def test_timeframe_duration_with_min_nb_journeys(self):
...
         query = ('journeys?from={_from}&'
                 'to={to}&'
                 'datetime={datetime}&'
                 'min_nb_journeys={min_nb_journeys}&'
                 'timeframe_duration={timeframe_duration}&').format( _from='stop_area:sa1',
                                                                     to='stop_area:sa3',
-                                                                    datetime="20180315T080000",
-                                                                    min_nb_journeys="21",
+                                                                    datetime='20180315T080000',
+                                                                    min_nb_journeys='21',
```
Comment (TeXitoi) [inline source/jormungandr/tests/journey_common_tests.py]: no quote

File: source/routing/raptor_api.cpp [hunk 3]
```diff
@@ -121,19 +115,19 @@ static bool keep_going(const uint32_t total_nb_journeys,
  * @brief Culling excessive journeys if we exceed limits
  */
 static void culling_excessive_journeys(const boost::optional<uint32_t>& min_nb_journeys,
-                                       const boost::optional<uint32_t>& timeframe_duration,
+                                       const boost::optional<DateTime>& timeframe_limit,
```
Comment (TeXitoi) [inline source/routing/raptor_api.cpp:118]: I think uint32_t is better that DateTime here.

File: source/routing/raptor_api.cpp [hunk 4]
```diff
@@ -121,19 +115,19 @@ static bool keep_going(const uint32_t total_nb_journeys,
...
-    if (timeframe_duration && journeys.size() > 0) {
+    if (timeframe_limit && journeys.size() > 0) {
 
         routing::JourneyCmp journey_cmp{clockwise};
         // filter journeys by end datetime until min_nb_journeys
         std::vector<routing::Journey> sorted{journeys.begin(), journeys.end()};
         std::sort(sorted.begin(), sorted.end(), journey_cmp);
 
-        auto is_inside = [clockwise](const routing::Journey& j, DateTime dt){
+        auto is_inside = [&clockwise](const routing::Journey& j, DateTime dt){
```
Comment (TeXitoi) [inline source/routing/raptor_api.cpp]: why?

File: source/routing/tests/routing_api_test.cpp [hunk 5]
```diff
@@ -3532,18 +3531,19 @@ BOOST_AUTO_TEST_CASE(journeys_with_time_frame_duration) {
     //-----------------------------------
     // Case 3 :
     // clockwise = false !!
-    // timeframe_duration = 10*60 (10 min)
     // timeframe_max_datetime = 86400 (24H)
     //
-    // We have a time frame duration = 20 min related to the first Journeys (08:30:00).
-    // The response must contains 2 journeys.
+    // timeframe_duration = 10*60*2 (20 min)
```
Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp]: 20*60?

File: source/routing/tests/routing_api_test.cpp [hunk 6]
```diff
@@ -3532,18 +3531,19 @@ BOOST_AUTO_TEST_CASE(journeys_with_time_frame_duration) {
     //-----------------------------------
     // Case 3 :
     // clockwise = false !!
-    // timeframe_duration = 10*60 (10 min)
     // timeframe_max_datetime = 86400 (24H)
```
Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp]: What's that?

File: source/routing/tests/routing_api_test.cpp [hunk 7]
```diff
@@ -3591,20 +3591,21 @@ BOOST_AUTO_TEST_CASE(journeys_with_time_frame_duration) {
...
-    // We have a time frame duration = 30 min related to the first Journeys (08:30:00) and
+    // We have a time frame duration = 1H related to the first Journeys (08:30:00) and
     // a min_nb_journeys = 20.
-    // We don't have 20 journeys in 30 min, so we continue until 20 journeys (min_nb_journeys is the main criteria).
-    // The response must contains 20 journeys.
+    // We don't have 20 journeys in 1H, so we continue until 20 journeys as
+    // min_nb_journeys is the limiting criteria.
+    // The response must contain 20 journeys.
     min_nb_journeys = 20;
     timeframe_duration = 60*60*1;  // 1H
```
Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp]: You can also correct that

File: source/routing/tests/routing_api_test.cpp [hunk 8]
```diff
@@ -3658,7 +3660,7 @@ BOOST_AUTO_TEST_CASE(journeys_with_time_frame_duration) {
                   sn_worker,
                   nt::RTLevel::Base,
                   2_min,
-                  60*60,
+                  timeframe_max_datetime,
```
Comment (TeXitoi) [inline source/routing/tests/routing_api_test.cpp]: That's not a datetime

**Review by TeXitoi — CHANGES_REQUESTED** [review]

File: source/jormungandr/tests/journey_common_tests.py [hunk 9]
```diff
@@ -1482,18 +1482,18 @@ def test_min_nb_transfers(self):
 
 @dataset({"min_nb_journeys_test": {}})
 class JourneysMinNbJourneys():
-    """
+    '''
```
Comment (mbounabi) [inline source/jormungandr/tests/journey_common_tests.py]: PEP 257 : For consistency, always use """triple double quotes"""
I won't put a comment on every changes but please revert this

**Review by mbounabi — CHANGES_REQUESTED** [review]

File: source/routing/raptor_api.cpp (hunk 3 above)
Comment (pbougue) [reply source/routing/raptor_api.cpp:118]: I do not agree (or I didn't understand why DateTime are for).
This is a DateTime in Navitia's semantics, not a duration, so I think it's better.

Comment (TeXitoi) [reply source/routing/raptor_api.cpp:118]: My bad.

**Review by xlqian — APPROVED** [review]

**Pull request merged**